Cannot create reproduction

### Link to the code that reproduces this issue

https://github.com/Vinnl/next-repro-csp-third-parties-bug

### To Reproduce

1. Run ` npx create-next-app -e reproduction-template`
2. Pick a name

### Current vs. Expected behavior

Instead of the repo, I got:

```
Creating a new Next.js app in /tmp/custom-properties-reproduction.

Downloading files for example reproduction-template. This might take a moment.

Installing packages. This might take a couple of minutes.

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: undefined@undefined
npm ERR! Found: react@18.3.1
npm ERR! node_modules/react
npm ERR!   react@"^18.2.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"19.0.0-beta-4508873393-20240430" from next@14.3.0-canary.54
npm ERR! node_modules/next
npm ERR!   next@"canary" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP PREEMPT_DYNAMIC Sat Apr 27 17:53:31 UTC 2024
  Available memory (MB): 15695
  Available CPU cores: 8
Binaries:
  Node: 22.1.0
  npm: 10.7.0
  Yarn: 1.22.19
  pnpm: 8.14.2
Relevant Packages:
  next: 14.2.3 // Latest available version is detected (14.2.3).
  eslint-config-next: 14.2.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.4.2
Next.js Config:
  output: N/A
```


### Which area(s) are affected? (Select all that apply)

create-next-app

### Which stage(s) are affected? (Select all that apply)

Other (Deployed)

### Additional context

(Note that I linked to an older reproduction to avoid this issue getting closed.)

--- a/examples/reproduction-template-pages/package.json
+++ b/examples/reproduction-template-pages/package.json
@@ -7,8 +7,8 @@
   },
   "dependencies": {
     "next": "canary",
-    "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react": "19.0.0-beta-4508873393-20240430",
+    "react-dom": "19.0.0-beta-4508873393-20240430"
   },
   "devDependencies": {
     "@types/node": "20.4.5",

--- a/examples/reproduction-template/package.json
+++ b/examples/reproduction-template/package.json
@@ -7,8 +7,8 @@
   },
   "dependencies": {
     "next": "canary",
-    "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react": "19.0.0-beta-4508873393-20240430",
+    "react-dom": "19.0.0-beta-4508873393-20240430"
   },
   "devDependencies": {
     "@types/node": "20.4.5",

